c2rust-transpile/src/translator/assembly.rs

```diff
@@ -722,10 +715,9 @@ impl<'c> Translation<'c> {
                 if let Ok(idx) = ref_str.parse::<usize>() {
                     outputs
                         .iter()
-                        .chain(inputs.iter())
+                        .chain(inputs)
                         .nth(idx)
-                        .map(operand_is_mem_only)
-                        .unwrap_or(false)
+                        .map_or(false, operand_is_mem_only)
```

Comment:
The previous way seems cleaner to me.

Reply:
It's certainly a matter of taste. However, the `map_or` function is already used 45 times in the project (10 times in transpiler, 35 in refactorer, and 1 in build script), and there is some value in shorter method chains (since there are less operations to parse in the head).

Reply:
I disagree.  You want longer method chains if each step in the chain is short.  That way you only need to keep one small step at a time in your head, instead of two at once.